eq-author-api/src/validation/customKeywords/validateRoutingRules.js

```diff
@@ -0,0 +1,73 @@
+const {
+  ERR_DESTINATION_DELETED,
+  ERR_DESTINATION_MOVED,
+} = require("../../../constants/validationErrorCodes");
+const availableRoutingDestinatinons = require("../../businessLogic/availableRoutingDestinations");
+const { some } = require("lodash");
+const newValidationError = (
+  keyword = "errorMessage",
+  dataPath = [],
+  message = null,
+  params = {}
+) => ({
+  keyword,
+  dataPath,
+  message,
+  params,
+});
+module.exports = function(ajv) {
+  ajv.addKeyword("validateRoutingRule", {
+    $data: true,
+    validate: function isValid(
+      otherFields,
+      entityData,
+      fieldValue,
+      dataPath,
+      parentData,
+      fieldName,
+      questionnaire
+    ) {
+      isValid.errors = [];
+      const splitDataPath = dataPath.split("/");
+      const currentPageId =
+        questionnaire.sections[splitDataPath[2]].pages[splitDataPath[4]].id;
+      const { sections, questionPages } = availableRoutingDestinatinons(
+        questionnaire,
+        currentPageId
+      );
+
+      if (
+        !entityData.destination.pageId &&
```

Comment:
Not necessary but can destructure entityData like
{ entityData: { destination: { pageId, sectionId, logical } } }